query-dsl/src/main/java/org/infinispan/query/dsl/Query.java

```diff
@@ -72,5 +85,10 @@
     * @param paramValues a Map of parameters
     * @return itself
     */
-   Query setParameters(Map<String, Object> paramValues);
+   Query<T> setParameters(Map<String, Object> paramValues);
+
+   /**
+    * @return the results of the query as an iterator.
+    */
+   CloseableIterator<T> iterator();
```

Comment:
Why do we need an ```iterator()``` method here and not in QueryResult ?

Reply:
```QueryResult``` is not lazy, but ```iterator()``` is.  I felt that mixing both in the same result class would feel strange

Reply:
I am just curious, but since iterator is blocking. Is it guaranteed that all results are available when invoking the iterator.next etc. or will it do paging (which would need to block)?

Reply:
> is it guaranteed that all results are available when invoking the iterator.next

No. It uses an underlying ```CacheStream``` for non-indexed and hybrid queries, or a ```org.infinispan.query.clustered.DistributedIterator``` in case the index can fully resolve the query. Both pulls chunks of items from remote nodes at a time (or will after ISPN-11881)

Reply:
Okay I was just wondering if this is a user facing API if we should have it be non blocking. But I guess we can always add a non blocking method later.

Reply:
It is user facing. Right now the focus was about unifying CacheQuery and Query, both are 100% blocking API. I started porting those things to Publisher, but stopped since ISPN 11 is just around the corner

Reply:
Okay, cool 👍